src/main/java/com/digitalasset/refapps/bondissuance/util/CommandsAndPendingSetBuilder.java

```diff
@@ -62,15 +62,15 @@ public void addCommand(Command cmd) {
       if (commands.isEmpty() && pendingContractIds.isEmpty()) {
         return Optional.empty();
       } else {
-        Instant time = timeManager.getTime();
         SubmitCommandsRequest commandsRequest =
             new SubmitCommandsRequest(
                 workflowId,
                 appId,
                 UUID.randomUUID().toString(),
                 party,
-                time,
-                time.plusSeconds(30),
+                Optional.empty(),
+                Optional.of(Duration.ofSeconds(30)),
```

Comment:
This should not be set, but because it was set, you discovered a bug in the ledger time handling. Thank you!
```suggestion
                Optional.empty(),
```